src/main/java/seedu/address/model/leave/Leave.java

```diff
@@ -0,0 +1,153 @@
+ * Represents a Leave request of an employee in the address book
+ */
+public class Leave {
+
+    private final Person employee;
+    private final String title;
+    private final String description;
+    private final LocalDate start;
+    private final LocalDate end;
+    private final String status;
+
+    /**
+     * Constructor for Leave object. Takes in a Person object, title and description, start and end date.
+     * Requires all fields to be non-null.
+     *
+     * @param employee
+     * @param title
+     * @param description
+     * @param start
+     * @param end
+     */
+    public Leave(Person employee, String title, LocalDate start, LocalDate end, String description)
+            throws EndBeforeStartException {
+
+        requireAllNonNull(employee, title, description, start, end);
+        if (end.isBefore(start)) {
+            throw new EndBeforeStartException();
+        }
+        this.employee = employee;
+        this.title = title;
+        this.description = description;
+        this.start = start;
+        this.end = end;
+        this.status = "Pending";
+    }
+
+    /**
+     * Constructor for Leave object without the optional description field.
+     *
+     * @param employee
+     * @param title
+     * @param start
+     * @param end
+     */
+    public Leave(Person employee, String title, LocalDate start, LocalDate end)
+            throws EndBeforeStartException {
+
+        requireAllNonNull(employee, title, start, end);
+        if (end.isBefore(start)) {
+            throw new EndBeforeStartException();
+        }
+        this.employee = employee;
+        this.title = title;
+        this.description = "";
+        this.start = start;
+        this.end = end;
+        this.status = "Pending";
+    }
+
+    public Person getEmployee() {
+        return employee;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getDescription() {
+        return description;
+    }
+
+    public LocalDate getStart() {
+        return start;
+    }
+
+    public LocalDate getEnd() {
+        return end;
+    }
+
+    public String getStatus() {
+        return status;
+    }
+
+    public boolean belongsTo(Person employee) {
+        return this.employee.equals(employee);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (o == this) {
+            return true;
+        }
+
+        if (!(o instanceof Leave)) {
+            return false;
+        }
+
+        Leave otherLeave = (Leave) o;
+        return otherLeave.getEmployee().equals(getEmployee())
+                && otherLeave.getTitle().equals(getTitle())
+                && otherLeave.getDescription().equals(getDescription())
+                && otherLeave.getStart().equals(getStart())
+                && otherLeave.getEnd().equals(getEnd())
+                && otherLeave.getStatus().equals(getStatus());
+    }
+
+    /**
+     * Returns true if both leaves have the same identity and data fields.
```

Comment:
What is the purpose of defining isSameLeave vs using the equals method?

Reply:
One is a strong equality, whereby all of the fields must be equal in order for two requests to be equal, while another one which we will more commonly be using, would be a bit more loose in the definition, as along the way we might choose to change a description, or approve/reject a particular leave request, and even if such attributes change they are essentially still the same leave request object.